code-challenges.test.js

```diff
@@ -15,47 +15,99 @@
+// the .split will convert a string into an array, .map will do something to each elelmet of the array and the .join will bring it back into a string
+}
+
+
 
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-const letterA = "a"
-// Expected output: ["Apple", "Banana", "Orange"]
-const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
 
+describe("particularletter", () => {
+  it("takes in an array of words and a single letter and returns all the words that contain that particular letter"), () => {
+    const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+    // const letterA = "a"
+    // // Expected output: ["Apple", "Banana", "Orange"]
+    const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+    // const letterE = "e"
+    // Expected output: ["Cherry", "Blueberry", "Peach"]
+    expect(particularletter(arrayOfWords1)).toEqual(["Apple", "Banana", "Orange"]),
+      expect(particularletter(arrayOfWords2)).toEqual(["Cherry", "Blueberry", "Peach"]);
+  });
+});
 
 
 // b) Create the function that makes the test pass.
 
+def arrayOfWords1(array, thing_to_find)
+  array.select { |word| word.include? thing_to_find }
+// .select will go through the array and select the matching input.   
+end
+
+
+
+
+
 
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-
+describe("fullhouse", () => {
+  it("takes in an array of 5 numbers and determines whether or not the array is a full house"), () => {
+    const hand1 = [5, 5, 5, 3, 3]
+    // // Expected output: true
+    const hand2 = [5, 5, 3, 3, 4]
+    // // Expected output: false
+    const hand3 = [5, 5, 5, 5, 4]
+    // Expected output: false
+    expect(fullhouse(hand1)).toEqual(true),
+    expect(fullhouse(hand2)).toEqual(false),
+    expect(fullhouse(hand3)).toEqual(false);
+  });
+});
 
 
 // b) Create the function that makes the test pass.
+
+
+
+const fullHouseOrNot = (arrayOfNums) => {
+  let occurances = 0;
+  if ([...new Set(arrayOfNums)].length == 2 && arrayOfNums.length === 5) {
+    arrayOfNums.forEach((num) =>
+      arrayOfNums[0] === num ? (occurances += 1) : occurances
+    );
+    return occurances == 2 || occurances == 3;
+  } else {
+    return false;
+  }
+};
```

Comment:
Your logic looks good. I would love to hear more about your process so I can understand how you got to this solution.